ksqldb-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java

```diff
@@ -173,6 +173,9 @@ public boolean isTopicExists(final String topic) {
           RetryBehaviour.ON_RETRYABLE.and(e -> !(e instanceof UnknownTopicOrPartitionException))
       );
       return true;
+    } catch (final TopicAuthorizationException e) {
+      LOG.debug("Topic {} has denied authorization. Marking topic as not existing.", topic);
```

Comment:
I originally had the code like this, but @rodesai convinced me that it shouldn't. If we know that we can't access it then we know that it exists. I feel like we should propagate this exception. Is there any issue you found that made you submit this PR?

Reply:
There is an RBAC muckracke test that is failing because it expects a user cannot see the topic when attempting to create a stream. But it fails now with an authorization exception on the topic.

I'm not sure what is the good solution, though, if letting fail with you don't have access to that topic, or as the old behavior which tell the user that topic does not exist. Perhaps we should do what Kafka does? If descripeTopics fails, then we should fail?

Reply:
that makes sense to me, though we should defer to security experts 😂

Reply:
@agavra I updated the method. It will throw a `KsqlTopicAuthorizationException` instead so the error message matches all the current authorization errors. i.e. `Authorization denied to Describe on topic(s): [foobar]`

I think this is better to do to avoid the below conflict:
```
> show topics;
t1

> create stream t2 with (kafka_topic='t2');
Topic 't2' not found.

> create stream t2 with (kafka_topic='t2', partitions=1, replicas=1);
Authorization denied to Create on topic(s): [foobar]
```
If the user has CREATE permissions on the cluster, then the above authorization message will still be displayed because the KSQL code assumes that the thrown `TopicAuthorizationException` from `KafkaTopicClientImpl.createTopic` method is due to lack of CREATE permissions. 

If we fix the previous issue, and add the real reason, like `Authorization denied to Describe on topic(s): [foobar]`, then what's the reason of using `Topic t2 not found.` when not using the partitions/replicas properties if users can get the real reason with it? 

I am leaving your fix but returning the right exception instead.